renderdoc/driver/gl/wrappers/gl_texture_funcs.cpp

```diff
@@ -1410,6 +1472,14 @@ bool WrappedOpenGL::Serialise_glTextureParameteriEXT(SerialiserType &ser, GLuint
   return true;
 }
 
+void WrappedOpenGL::MarkWriteReferencedInBackground(GLResourceRecord *record, FrameRefType refType)
+{
+  if(!record || !IsBackgroundCapturing(m_State))
+    return;
+
+  GetResourceManager()->MarkResourceFrameReferenced(record->GetResourceID(), refType);
+}
```

Comment:
This function is a bit out of place in the middle of everything here. It would be better placed in `gl_driver.cpp` or at the top of this file.

Reply:
Actually going back to this the logic is also backwards/wrong - the critical thing for getting write references before the GL call is that we need them to happen before the GL call while _active_ capturing. In background the ordering is not important as long as we know something has been written, but while actively capturing we need to know that it's about to happen before the GL call so we can snapshot the initial contents when needed.